apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -48,25 +48,31 @@ namespace Google.Devtools.AspNet
     /// </remarks>
     public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
     {
+        // The max number of tasks that will be allowed when waiting for the client task.
+        internal const int MaxWaitingTasks = 100;
+
         // The formated Google Cloud Platform project id.
         private readonly string _projectResourceName;
 
         // The service context in which this error has occurred.
         // See: https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events#ServiceContext
         private readonly ServiceContext _serviceContext;
 
-        private readonly ReportErrorsServiceClient _client;
+        private readonly Task<ReportErrorsServiceClient> _clientTask;
+
+        // The number tasks that have been created while waiting for the client task.
+        private int taskCounter = 0;
 
         /// <summary>
         /// Creates an instance of <see cref="CloudErrorReportingExceptionLogger"/>
         /// </summary>
-        /// <param name="client">The Error Reporting client.</param>
+        /// <param name="clientTask">The Error Reporting client.</param>
         /// <param name="projectId">The Google Cloud Platform project ID.</param>
         /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
         /// <param name="version">Represents the source code version that the developer provided.</param> 
         public static CloudErrorReportingExceptionLogger Create(
```

Comment:
It might be worth having an overload taking an actual `ReportErrorsServiceClient` just for simplicity. (That overload can use Task.FromResult.)